### DISCUSSION TITLE
Cant get $extends to working with $transaction

### DISCUSSION THREAD

**Discussion by GZLiew** [prisma/prisma#20718] [Q&A]
### Question

Cant get $extends to working with $transaction

```ts
    prisma.$extends({
      client: {
        async $transaction<T>(handler: () => Promise<T>) {
          return Prisma.getExtensionContext(prisma).$transaction(
            async (tx: any) => {
              if (!prismaService.getTransaction()) {
                prismaService.setTransaction(tx);
              }

              try {
                const result = await handler();
                return result;
              } finally {
                console.log("DID I GET CALLED?");
                this.clsService.set("prismaTransaction", undefined);
              }
            }
          );
        },
      },
      query: {
        $allModels: {
          $allOperations({ model, operation, args, query }) {
            const tx = prismaService.getTransaction();

            if (tx) {
              return tx?.[model]?.[operation]?.(args);
            }

            return query(args);
          },
        },
      },
    });
```

**Comment by ludralph** [comment]
Hello @GZLiew 👋 

What intended behavior are you trying to achieve using `prisma.$extends` and `$transaction`? Are you having any specific error messages when using the provided code? Can you please describe the purpose and functionality of `prismaService.getTransaction()` and `prismaService.setTransaction(tx)`?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
